payment-processor/src/main/kotlin/org/ostelco/prime/paymentprocessor/StripePaymentProcessor.kt

```diff
@@ -223,63 +265,75 @@ class StripePaymentProcessor : PaymentProcessor {
             }
 
-    private fun subscriptionStatus(subscription: Subscription): Triple<PaymentStatus, String, String> {
+    private fun getSubscriptionPaymentInfo(subscription: Subscription): SubscriptionPaymentInfo {
         val invoice = subscription.latestInvoiceObject
         val intent = invoice?.paymentIntentObject
 
-        return when (subscription.status) {
-            "active", "incomplete" -> {
-                if (intent != null)
-                    Triple(when (intent.status) {
+        return if (subscription.status == "trialing") {
+            PaymentStatus.TRIAL_START
+        } else {
+            require(intent != null) {
+                "Expected an invoice intent with invoice ${invoice.id} for subscription ${subscription.id} " +
+                        "with status ${subscription.status}"
+            }
+            when (subscription.status) {
+                "active" -> {
+                    when (intent.status) {
                         "succeeded" -> PaymentStatus.PAYMENT_SUCCEEDED
+                        else -> throw IllegalArgumentException("Unexpected intent ${intent.status} for invoice ${invoice.id} " +
+                                "for subscription ${subscription.id} with status ${subscription.status}")
+                    }
+                }
+                "incomplete" -> {
+                    when (intent.status) {
                         "requires_payment_method" -> PaymentStatus.REQUIRES_PAYMENT_METHOD
                         "requires_action" -> PaymentStatus.REQUIRES_ACTION
-                        else -> throw RuntimeException(
-                                "Unexpected intent ${intent.status} for Stipe payment invoice ${invoice.id}")
-                    }, invoice.id, invoice.charge)
-                else {
-                    throw RuntimeException(
-                            "'Intent' absent in response when creating subscription ${subscription.id} with status ${subscription.status}")
+                        else -> throw IllegalArgumentException("Unexpected intent ${intent.status} for invoice ${invoice.id} " +
+                                "for subscription ${subscription.id} with status ${subscription.status}")
+                    }
+                }
+                else -> {
+                    throw IllegalArgumentException(
+                            "Got unexpected status ${subscription.status} when creating subscription ${subscription.id}")
                 }
             }
-            "trialing" -> {
-                Triple(PaymentStatus.TRIAL_START, "", "")
-            }
-            else -> {
-                throw RuntimeException(
-                        "Got unexpected status ${subscription.status} when creating subscription ${subscription.id}")
-            }
+        }.let {
+            SubscriptionPaymentInfo(id = subscription.id,
+                    status = it,
+                    invoiceId = if (invoice != null)
```

Comment:
consider using either braces in branches of if/else or wrapping it in a () (or both).